### PR TITLE
Use DI Get_it

### LINKED ISSUE #58: Use DI Get_it in project
- [x] Research how to use Get_it (find examples)
- [ ] Use Get_it and injectable

______________

This issue size:
- Estimate SP: 3
- Actual SP: ?

### PR DESCRIPTION
**Pull request by Yura-0**
Resolve #58

### CODE REVIEW

File: Sources/Userlog/userlog/lib/injection.config.dart [hunk 1]
```diff
@@ -0,0 +1,36 @@
+// GENERATED CODE - DO NOT MODIFY BY HAND
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/injection.config.dart:1]: What do you think, maybe it is better to note this file in gitignore?

File: Sources/Userlog/userlog/lib/services/log/services/log_service.dart [hunk 2]
```diff
@@ -1,9 +1,11 @@
 import 'dart:async';
 import 'dart:math';
 
+import 'package:injectable/injectable.dart';
 import 'package:userlog/services/log/api/i_log_service.dart';
 import 'package:userlog/services/log/api/models/record.dart';
 
+@singleton
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/services/log/services/log_service.dart:8]: If it is possible, dont mark this clas as singleton here

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/lib/injection.config.dart (hunk 1 above)
Comment (Yura-0) [reply Sources/Userlog/userlog/lib/injection.config.dart:1]: Okay, but I must add a new recommendation in userlog README file, where I explain how to generate dependencies

File: Sources/Userlog/userlog/lib/injection.config.dart [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/injection.config.dart:1]: It is still not hidden

File: Sources/Userlog/userlog/lib/services/log/dependencies/log_service_module.dart [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import 'package:injectable/injectable.dart';
+
+import 'package:userlog/services/log/api/i_log_service.dart';
+import 'package:userlog/services/log/services/log_service.dart';
+
+@module
+abstract class LogServiceModule {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/lib/services/log/dependencies/log_service_module.dart:7]: Renae it to LogServiceDiModule

File: Sources/Userlog/userlog/scripts/build.dart [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// Generate file injection.confic.dart in lib folder and start build in debug mode
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/scripts/build.dart:1]: it has to be scripts in yaml file

like here https://flutterawesome.com/define-and-use-scripts-from-your-pubspec-yaml-file/

**Review by sahlet-official — CHANGES_REQUESTED** [review]

**Review by sahlet-official — CHANGES_REQUESTED** [review]

File: Sources/Userlog/userlog/scripts/build.dart [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+void main(List<String> args) {
+  const buildRunnerCmd =
+      'flutter packages pub run build_runner build --delete-conflicting-outputs';
+  const buildCmd = 'flutter build apk --debug';
+
+  Process.runSync(buildRunnerCmd, [], runInShell: true);
+
+  final processResult = Process.runSync(buildCmd, [], runInShell: true);
+
+  if (processResult.exitCode == 0) {
```
Comment (sahlet-official) [inline Sources/Userlog/userlog/scripts/build.dart:14]: we dont need this code at all, we only need ti run the script. all other will be printed on the screen

**Review by sahlet-official — CHANGES_REQUESTED** [review]

**Pull request merged**